# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Realm.cpp
# Commit: 687d6e55be66
# Full Hash: 687d6e55be668914ce8f59769d06bedbf1736406
# Author: Matthew Gaudet <mgaudet@mozilla.com>
# Date: 2025-06-03 03:42:51
# Description:
#   Bug 1969445 - Don't report OOM telemetry if the associated global isn't done setting up r=mccr8,jandem
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D251924
# ==============================================================================

diff -r 43ee34b22c20 -r 687d6e55be66 js/src/vm/Realm.cpp
--- a/js/src/vm/Realm.cpp	Mon Jun 02 19:24:07 2025 +0000
+++ b/js/src/vm/Realm.cpp	Mon Jun 02 19:24:12 2025 +0000
@@ -653,6 +653,10 @@
   return realm->realmPrivate();
 }
 
+JS_PUBLIC_API bool JS::HasRealmInitializedGlobal(JS::Realm* realm) {
+  return realm->hasInitializedGlobal();
+}
+
 JS_PUBLIC_API void JS::SetRealmPrivate(JS::Realm* realm, void* data) {
   realm->setRealmPrivate(data);
 }